source/administration/concepts.rst

```diff
@@ -0,0 +1,158 @@
+      2020-01-03-MinIO-Advanced-Deployment.png
+      MinIO-Logo.png
+   /videos/
+      2020-01-04-MinIO-Interview.mp4
+   /articles/
+      /john.doe/
+         2020-01-02-MinIO-Object-Storage.md
+         2020-01-02-MinIO-Object-Storage-comments.json
+      /jane.doe/
+         2020-01-03-MinIO-Advanced-Deployment.png
+         2020-01-02-MinIO-Advanced-Deployment-comments.json
+         2020-01-04-MinIO-Interview.md
+
+MinIO supports multiple levels of nested directories and objects to support even the most dynamic object storage workloads.
+
+
+Encryption
+----------
+
+Encryption-at-Rest
+   A method of data security that stores objects on a disk in an encoded state.
+   To read the data, the server must first decrypt the data using a decryption key.
+
+Encryption-in-Transit
+   A method of data security that encrypts objects as they transmit from one location to another.
+   Modern data transfer utilizes the TLS security standard to secure data during transmission.
+
+TLS
+   The Internet Engineering Task Force (IETF) maintains the security protocol called **Transport Layer Security (TLS)**.
+   TLS provides the standards used by internet communications to support encryption, authentication, and data integrity through a TLS handshake.
+   The handshake confirms the identify of the requesting party and whether the party should have access to the requested data.
+
+   After confirming identity and access, TLS provides the cipher to encrypt and then decrypt the transfer of information from the server to the requesting client.
+
+   TLS succeeds and replaces the previously used Secure Socket Layer (SSL)
+
+   MinIO supports versions 1.2 and 1.3 of the TLS standard.
+
+
+Object Management
+-----------------
+
+:ref:`Information Lifecycle Management (ILM) <minio-lifecycle-management>`
+   Information Lifecycle Management establishes rules to determine where to retain an object and how long to retain it.
+   For exaple, some objects may need to be retained for legal reasons for a specific period of time before anyone can delete them.
+   Other objects should be deleted after three days.
+   Objects that are accessed frequently should go in warm storage while those needed only for archival purposes and rarely accessed can move to cold storage.
+
+   ILM is the set of rules that determine when to move objects between storage classes and how long to retain objects or prevent their deletion.
+
+Locking
+   MinIO uses a method of object retention called Object Locking.
+   This Write Once Read Many (WORM) mechanism protects an object's immutability, preventing any changes to an object while in a locked state.
+
+   Locks can be based on a specified period of time for unprivileged users (governance mode) or for all users (compliance mode).
+   Administrations can use a legal hold to retain objects indefinitely, which prevents all users from deleting or modifying the locked object.
+
+Prefix
+   Prefixes are the beginning part of an object name.
```

Comment:
Hmm, not really.
Per Amazon's docs, S3 is a flat structure with no real hierarchy whatsoever.
Folders are simply an idea construed from a common string at the beginning of the object name with a specified delimiter character. The common string (the prefix) is, indeed, part of the object's _key name_.

See https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-folders.html and also https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html.

> In Amazon S3, buckets and objects are the primary resources, and objects are stored in buckets. **Amazon S3 has a flat structure instead of a hierarchy like you would see in a file system.** However, for the sake of organizational simplicity, the Amazon S3 console supports the folder concept as a means of grouping objects. **It does this by using a shared name prefix for objects (that is, objects have names that begin with a common string).** Object names are also referred to as key names.

A "folder" is nothing more than a 0 byte object added for the convenience of the user's mental model that comes from our long history of traditional file systems.